src/main/java/supercoder79/survivalgames/game/map/SurvivalGamesChunkGenerator.java

```diff
@@ -58,268 +58,278 @@
-			for (int x = chunkX; x < chunkX + 16; x++) {
-				for (int z = chunkZ; z < chunkZ + 16; z++) {
-					int y = world.getTopY(Heightmap.Type.MOTION_BLOCKING, x, z);
-					mutable.set(x, y, z);
-
-					BlockState down = world.getBlockState(mutable.down());
-					if (down.isOpaqueFullCube(world, mutable.down()) || down.isIn(BlockTags.LEAVES)) {
-						world.setBlockState(mutable, Blocks.SNOW.getDefaultState(), 3);
-
-						if (down.contains(Properties.SNOWY)) {
-							world.setBlockState(mutable.down(), down.with(Properties.SNOWY, true), 3);
-						}
-					} else if (down.getFluidState().isIn(FluidTags.WATER)) {
-						world.setBlockState(mutable.down(), Blocks.ICE.getDefaultState(), 3);
-					}
-				}
-			}
-		}
-	}
-
-	@Override
-	public void populateBiomes(Registry<Biome> registry, Chunk chunk) {
-		ChunkPos chunkPos = chunk.getPos();
-		((ProtoChunk)chunk).setBiomes(new BiomeArray(registry, chunkPos, this.biomeSource));
-	}
-
-	@Override
-	public void carve(long seed, BiomeAccess access, Chunk chunk, GenerationStep.Carver carver) {
-	}
-
-
-}
+        List<SurvivalGamesJigsawGenerator> generators = new ArrayList<>();
+
+        this.noiseGenerator.initialize(random, config);
+        if(config.dimension.getPath().equals("the_nether")) {
+            this.defaultState = Blocks.NETHERRACK.getDefaultState();
+            this.defaultFluid = Blocks.LAVA.getDefaultState();
+        } else {
+            this.defaultState = Blocks.STONE.getDefaultState();
+            this.defaultFluid = Blocks.WATER.getDefaultState();
+        }
+
+        ChunkMask mask = new ChunkMask();
+        ChunkBox townArea = new ChunkBox();
+
+        if (config.townDepth > 0) {
+            SurvivalGamesJigsawGenerator generator = new SurvivalGamesJigsawGenerator(server, this, piecesByChunk);
+            generator.arrangePieces(new BlockPos(0, 64, 0), new Identifier("survivalgames", "starts"), config.townDepth);
+            townArea = generator.getBox();
+            generators.add(generator);
+            mask.and(townArea);
+        }
+
+        for (int i = 0; i < config.outskirtsBuildingCount; i++) {
+            int startX = random.nextInt(config.borderConfig.startSize / 2) - random.nextInt(config.borderConfig.startSize / 2);
+            int startZ = random.nextInt(config.borderConfig.startSize / 2) - random.nextInt(config.borderConfig.startSize / 2);
+
+            if (!this.noiseGenerator.shouldOutskirtsSpawn(startX, startZ)) {
+                continue;
+            }
+
+            BlockPos start = new BlockPos(startX, 0, startZ);
+            ChunkPos chunkPos = new ChunkPos(start);
+
+            if (mask.isIn(chunkPos)) {
+                continue;
+            }
+
+            SurvivalGamesJigsawGenerator outskirtGenerator = new SurvivalGamesJigsawGenerator(server, this, piecesByChunk);
+            outskirtGenerator.arrangePieces(start, new Identifier("survivalgames", config.dimension.getPath() + "_outskirts_buildings"), 0);
```

Comment:
Ideally we would have a custom codec entry for the outskirts path. That'll allow us to eventually add tropical buildings to tropical maps for example